### ISSUE TITLE
Error: Cannot find module 'fuge'

### ISSUE THREAD

**Issue by pedx78** [apparatus/fuge#74]
going through example on this site. (OSX 10.11)

`fuge generate system`
`fuge shell ./fuge/compose-dev.yml`

However, when I run 'start all', I get following error : 

```

? fuge> start all
running: frontend
running:  PROXY_HOST=127.0.0.1 SERVICE_HOST=0.0.0.0 frontend_PORT=10000 api_PORT=10001 service1_PORT=10002 service2_PORT=10003 SERVICE_PORT=20000  exec /usr/local/Cellar/node/5.8.0/bin/node -r fuge index.js
running: api
running:  PROXY_HOST=127.0.0.1 SERVICE_HOST=0.0.0.0 frontend_PORT=10000 api_PORT=10001 service1_PORT=10002 service2_PORT=10003 SERVICE_PORT=20001  exec /usr/local/Cellar/node/5.8.0/bin/node -r fuge index.js
running: service1
running:  PROXY_HOST=127.0.0.1 SERVICE_HOST=0.0.0.0 frontend_PORT=10000 api_PORT=10001 service1_PORT=10002 service2_PORT=10003 SERVICE_PORT=20002  exec /usr/local/Cellar/node/5.8.0/bin/node -r fuge service.js
running: service2
running:  PROXY_HOST=127.0.0.1 SERVICE_HOST=0.0.0.0 frontend_PORT=10000 api_PORT=10001 service1_PORT=10002 service2_PORT=10003 SERVICE_PORT=20003  exec /usr/local/Cellar/node/5.8.0/bin/node -r fuge service.js
running: __proxy
? fuge> [frontend - 2261]: module.js:341
    throw err;
    ^

Error: Cannot find module 'fuge'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at module.js:504:12
    at Array.forEach (native)
    at Function.Module._preloadModules (module.js:503:12)
    at Function.startup.preloadModules (node.js:826:38)
    at startup (node.js:119:17)
    at node.js:933:3
[api - 2262]: module.js:341
    throw err;
    ^

Error: Cannot find module 'fuge'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at module.js:504:12
    at Array.forEach (native)
    at Function.Module._preloadModules (module.js:503:12)
    at Function.startup.preloadModules (node.js:826:38)
    at startup (node.js:119:17)
    at node.js:933:3
process exit [2261]: frontend
process exit [2262]: api
[service2 - 2264]: module.js:341
    throw err;
    ^
```

**Comment by AdrianRossouw** [comment]
did you install fuge globally?

**Comment by pedx78** [comment]
Yes, I already installed fuge globally.
`npm install -g fuge`

I solved issue by going to each of the generated services, and manually install fuge locally, and it seems to be running now.

**Comment by AdrianRossouw** [comment]
that really shouldn't be neccesary tho.

@geek could this be based on those changes you made recently in https://github.com/apparatus/fuge-runner/pull/3/files

There's another PR related to this too : https://github.com/apparatus/fuge-runner/pull/12

Thinking about it, it's actually a bit weird that the processes you run require on the code that is running the processes. That strikes me as a infinite-loopy, and something that can lead to circular dependencies.

**Comment by geek** [comment]
@AdrianRossouw definitely looks related.  It shouldn't cause any circular reference issues, you are spawning a process that will preload fuge... even if the module itself requires fuge, fuge won't be requiring the module when fuge is loaded (confusing, but you know what I mean).

The issue is likely that we are assuming fuge to be in the global module path to the node binary found by just running node.  I will offer up a PR

**Comment by AdrianRossouw** [comment]
i feel really weird about requiring the service running in the app to require all of fuge.
can we split the bits needed for proccess communication into a separate module, that they can both include.

Another thing to keep in mind is that we might not even be able to guarantee that the service is in the path of the system's node_modules. IE: the service could be a sibling directory to the service.

so how about `-r <absolute path>` ?

**Comment by geek** [comment]
@pedx78 I have a PR for a fix that I have verified solves this issue

**Comment by AdrianRossouw** [comment]
I have released 0.6.2 with the fix for this.
